### PR TITLE
First Steps Towards Replacing sqlite_orm with sqlite_modern_cpp

### PR DESCRIPTION
**Pull request by irq0**
Add `sqlite_modern_cpp` alongside `sqlite_orm` as a submodule. Wrap
include in `dbapi.h` to work around it's namespace `sqlite` coliding
with ours. Convert `SQLiteList::versions` and add type mappers for
`ceph::real_time`, `ObjectState` and `VersionType`

Epic: https://github.com/aquarist-labs/s3gw/issues/742
Issue: https://github.com/aquarist-labs/s3gw/issues/788
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

### CODE REVIEW

**Review by jecluis — COMMENTED** [review]
lgtm - but lets only merge this after #233

**Review by 0xavi0 — DISMISSED** [review]
lg2m.

I have ready the changes for all blobs in a generic way using this branch as the base, so I will push my PR after this is merged.
I have also the whole Users queries converted to the new sqlite library.

**Comment by jecluis** [comment]
@irq0 we have a conflict. Lets address that so we can merge this asap, so we can avoid recurring conflicts.

**Review by 0xavi0 — APPROVED** [review]

**Pull request merged**
